### PR TITLE
Add build command to dock-cli

### PR DESCRIPTION
**Pull request by Elfiggo**
Allows a user to successfully docker-compose build a container and reset it straight away.

### CODE REVIEW

**Comment by sroze** [comment]
Can you update the documentation too?

File: src/Cli/BuildCommand.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    private $projectManager;
+
+    /**
+     * @var Project
+     */
+    private $project;
+
+    /**
+     * @param DockerComposeProjectManager $projectManager
+     * @param Project        $project
```
Comment (sroze) [inline src/Cli/BuildCommand.php]: If you are using docblocks, then use `php-cs-fixer` my friend :wink:

File: src/Project/DockerComposeProjectManager.php [hunk 2]
```diff
@@ -91,4 +91,22 @@ public function reset(Project $project, array $containers = [])
...
+    {
+        $composePath = $this->composeExecutableFinder->find();
+
+        $this->processRunner->run(implode(' && ', array_map(function ($action) use ($composePath, $containers) {
+            return implode(' ', [
+                $composePath,
+                $action,
+                implode(' ', $containers),
+            ]);
+        }, ['build'])));
```
Comment (sroze) [inline src/Project/DockerComposeProjectManager.php]: I would remove this useless `array_map`, you simply ends with:

``` php
$this->processRunner->run(implode(' ', [
    $composePath,
    'build',
    implode(' ', $containers),
]));
```

File: src/Cli/BuildCommand.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+     */
+    public function __construct(DockerComposeProjectManager $projectManager, Project $project)
```
Comment (sroze) [inline src/Cli/BuildCommand.php]: I believe that you don't want to typehint `DockerComposeProjectManager` here, but your interface `ProjectBuildManager`. Then, you'll have to add this interface to the project managers used for tests to fix the tests :)

File: README.md [hunk 4]
```diff
@@ -45,6 +45,7 @@ The following commands are available:
 - [`logs` command](docs/cmd-logs.md) follows the logs of all or a given container.
 - [`run` command](docs/cmd-run.md) run a command in a container with the application context.
 - [`reset` command](docs/cmd-reset.md) kill, remove and start again a container.
+- [`build` command](docs/cmd-build.md) starts the Docker environment for the project.
```
Comment (sroze) [inline README.md]: It's actually building the Docker images and resetting the containers, nop?

File: app/container.php [hunk 5]
```diff
@@ -143,6 +144,9 @@ function (ConsoleCommandEvent $event) use ($userInteraction) {
 $container['command.stop'] = function ($c) {
     return new StopCommand($c['project.manager'], $c['compose.project']);
 };
+$container['command.build'] = function ($c) {
+    return new BuildCommand($c['project.manager.docker_compose'], $c['compose.project']);
```
Comment (sroze) [inline app/container.php:148]: You probably want to either inject `project.manager` or an other service (that can/should be an alias) called `project.build_manager`.

File: src/Cli/BuildCommand.php (hunk 3 above)
Comment (Elfiggo) [inline src/Cli/BuildCommand.php]: Yeah, my bad. worked on my machine ;)

File: docs/cmd-build.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# `build` command
+
+This command builds and resets a set of containers. It will run `docker-compose build [containers]` and then 
+run the reset command against those containers.
+
+That's extremely useful when the configuration of the running process needs to be updated for
+instance, as building the container will not restart it and it won't load the new configuration if the process to not support runtime
```
Comment (sroze) [inline docs/cmd-build.md]: I would remove "if the process _do_ not support runtime", as there's few issues around the watchers in :docker: containers.

**Comment by sroze** [comment]
This will fix #78.

**Comment by sroze** [comment]
Thank you @Elfiggo !

**Pull request merged**